Dockerfile

```diff
@@ -1,99 +1,131 @@
 RUN apt update
 RUN apt upgrade -y
-RUN apt install postgresql-client-14 postgresql-14 postgresql-14-cron pgbouncer postgresql-server-dev-14 libcurl4-openssl-dev git make g++ locales-all -y
-
+RUN apt install postgresql-client-14 postgresql-14 postgresql-14-cron pgbouncer postgresql-server-dev-14 libcurl4-openssl-dev -y
 RUN git clone --depth 1 https://github.com/pramsey/pgsql-http
 RUN cd pgsql-http && make && make install
 
-# Install NodeJS
+## Install NodeJS
 RUN ["bash", "-c", "curl -fsSL https://deb.nodesource.com/setup_16.x | bash -"]
 RUN apt install nodejs -y
 RUN npm install -g yarn
 
+#####################################################################################################################################################
+
 # Run service
 FROM base AS run-stage
 
 WORKDIR /run
 
+## Copy files
 COPY dashboard/run/tsconfig.json tsconfig.json
 COPY dashboard/run/src src
 
+## Install stage dependencies
 COPY dashboard/run/package.json dashboard/run/yarn.lock ./
 RUN yarn install
 
+## Build
 RUN yarn build
+
+## Prune dev dependencies
 RUN yarn install --production
 
+#####################################################################################################################################################
+
 # Dashboard
 FROM base AS dashboard-stage
 
 WORKDIR /dashboard
 
+## Copy files
 COPY dashboard/tsconfig.json dashboard/tailwind.config.js dashboard/craco.config.js ./
 COPY dashboard/public public
 COPY dashboard/src src
 
+## Install stage dependencies
 COPY dashboard/package.json dashboard/yarn.lock ./
 RUN yarn install
 
+## Default ENVs that can be overwritten
 ARG IASQL_ENV=local
 ENV REACT_APP_IASQL_ENV=$IASQL_ENV
 ENV GENERATE_SOURCEMAP=false
+
+## Build
 RUN yarn build
+
+## Prune dev dependencies
 RUN yarn install --production
 
+#####################################################################################################################################################
+
 # Engine
 FROM base AS engine-stage
 
 WORKDIR /engine
 
-COPY docker-entrypoint.sh ormconfig.js tsconfig.json ./
+## Copy files
+COPY ormconfig.js tsconfig.json ./
 COPY src src
 
+## Install stage dependencies
 COPY package.json yarn.lock ./
 RUN yarn install
 
+## Build
 RUN yarn build
+
+## Prune dev dependencies
 RUN yarn install --production
 
+#####################################################################################################################################################
+
 # Main stage
 FROM base AS main-stage
 
-COPY ./src/scripts/postgresql.conf /etc/postgresql/14/main/postgresql.conf
-COPY ./src/scripts/pg_hba.conf /etc/postgresql/14/main/pg_hba.conf
-
+## Copy from run-stage
 WORKDIR /dashboard/run
 COPY --from=run-stage /run/package.json package.json
 COPY --from=run-stage /run/node_modules node_modules
 COPY --from=run-stage /run/dist dist
 
+## Copy from dashboard-stage
 WORKDIR /dashboard
 COPY --from=dashboard-stage /dashboard/build build
```

Comment:
When you run this and connect to localhost:8888 does it work? I would expect it not to with how it is structured.

Reply:
what do you mean? It works and all tests have passed. also, nothing has changed in this particular line from yesterday?

Reply:
Oh, because the way the run service hosts the dashboard build artifact assumes it's located in `../build` but now it needs to be in `../dashboard/build` so I honestly don't understand how that part would be working?

Reply:
I made sure that was taken into account when structuring the directories in the container. The run service code runs under `dashboard/run` and it's `index.ts` is in `dashboard/run/dist` directory and the relative path in the run service to host the dashboard is `../../build`, so the first `..` goes back to `dashboard/run` and the second to `dashboard` where we have the `build` directory.

Reply:
Ahh, I see now, the `WORKDIR` is `/dashboard/run`. I saw all of the `COPY --from=run-stage /run/...` and thought it was also put in the top-level.